sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -211,6 +211,55 @@ object JdbcUtils extends Logging {
   }
 
   /**
+   * Returns the schema if the table already exists in the JDBC database.
+   */
+  def getSchema(conn: Connection, url: String, table: String): Option[StructType] = {
+    val dialect = JdbcDialects.get(url)
+
+    try {
+      val statement = conn.prepareStatement(dialect.getSchemaQuery(table))
+      try {
+        Some(getSchema(statement.executeQuery(), dialect))
+      } catch {
+        case _: SQLException => None
+      } finally {
+        statement.close()
+      }
+    } catch {
+      case _: SQLException => None
+    }
+  }
+
+  /**
+   * Returns a schema using rddSchema's column sequence and tableSchema's column names.
+   *
+   * When appending data into some case-sensitive DBMSs like PostgreSQL/Oracle, we need to respect
+   * the existing case-sensitive column names instead of RDD column names for user convenience.
+   * See SPARK-18123 for more details.
+   */
+  def normalizeSchema(
+      rddSchema: StructType,
+      tableSchema: StructType,
+      caseSensitive: Boolean): StructType = {
```

Comment:
Then, do you want to create `rddSchema` StructField with tableSchema's column name (if not matched)?